rust/src/encodings/plain.rs

```diff
@@ -99,9 +98,13 @@ impl<'a> PlainEncoder<'a> {
             self.encode_boolean(boolean_arr).await?;
         } else {
             let byte_width = array.data_type().byte_width();
-            self.writer
-                .write_all(&data.buffers()[0].as_slice()[0..array.len() * byte_width])
-                .await?;
+            let slice = unsafe {
+                from_raw_parts(
```

Comment:
does this need to be done for boolean as well?

Reply:
Boolean needs to shift to the byte array,thus have to copy the data to new buffer, so this does not work for boolean.